go-controller/pkg/config/utils.go

```diff
@@ -84,9 +84,15 @@ func ParseClusterSubnetEntries(clusterSubnetCmd string) ([]CIDRNetworkEntry, err
 // These entries define the flow collectors OVS will send flow metadata by using NetFlow/SFlow/IPFIX.
 func ParseFlowCollectors(flowCollectors string) ([]HostPort, error) {
 	var parsedFlowsCollectors []HostPort
-
+	readCollectors := map[string]struct{}{}
 	collectors := strings.Split(flowCollectors, ",")
 	for _, v := range collectors {
+		if _, ok := readCollectors[v]; ok {
+			// duplicate flow collector. Ignore it
```

Comment:
Will this work without parsing the addresses?  Because OVS probably parses into binary form.

IPv6 has many equivalent forms with different text representations.  We probably can't ever check the addresses fully and may need to parse the OVS errors.

e.g.
```
fd00:1101:0000:0001:0000:0000:0000:0002
fd00:1101:0000:0001::0002
```


Not sure if OVS accepts IPv4 with leading `0` either.  `192.168.001.010`

Reply:
Curious

```
sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@ipfix create ipfix targets='["10.128.02.12:2055","10.128.2.12:2055"]' cache_active_timeout=60 -- set bridge br-int ipfix=@ipfix
19401d8d-c7e0-4e1b-8757-a89f6fa954f4
sh-4.4# ovs-vsctl find ipfix
_uuid               : 19401d8d-c7e0-4e1b-8757-a89f6fa954f4
cache_active_timeout: 60
cache_max_flows     : []
external_ids        : {}
obs_domain_id       : []
obs_point_id        : []
other_config        : {}
sampling            : []
targets             : ["10.128.02.12:2055", "10.128.2.12:2055"]
sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@ipfix create ipfix targets='["10.128.2.12:2055","10.128.2.12:2055"]' cache_active_timeout=60 -- set bridge br-int ipfix=@ipfix
ovs-vsctl: ["10.128.2.12:2055","10.128.2.12:2055"]: set contains duplicate value

sh-4.4# /usr/bin/ovs-vsctl --timeout=15 -- --id=@ipfix create ipfix targets='["[fd00:1101:0000:0001:0000:0000:0000:0002]:2055","[fd00:1101:0000:0001::0002]:2055"]' cache_active_timeout=60 -- set bridge br-int ipfix=@ipfix
94da95ff-d5fa-4d32-9368-1b9762bebfcd
sh-4.4# ovs-vsctl find ipfix
_uuid               : 94da95ff-d5fa-4d32-9368-1b9762bebfcd
cache_active_timeout: 60
cache_max_flows     : []
external_ids        : {}
obs_domain_id       : []
obs_point_id        : []
other_config        : {}
sampling            : []
targets             : ["[fd00:1101:0000:0001:0000:0000:0000:0002]:2055", "[fd00:1101:0000:0001::0002]:2055"]
```

So I guess this is okay.  OVS accepts whatever and is just checking text forms?

Reply:
I checked that the `net.ParseIP` function that is extensively used in OVN-Kubernetes neither accepts zero-leaded IPV4 addresses, so I guess anyway there is no risk to pass a zero-leaded IP to OVS.

On the other side, I have committed a new version that discards equivalent IPs even if they are passed with different texts.